Subscriber/BasicSettingsTaxSubscriber.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace SwagTax\Subscriber;
+
+use Doctrine\DBAL\Connection;
+use Enlight\Event\SubscriberInterface;
+
+class BasicSettingsTaxSubscriber implements SubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            'Enlight_Controller_Action_PreDispatch_Backend_Config' => 'onPreDispatch'
+        ];
+    }
+
+    public function onPreDispatch(\Enlight_Controller_ActionEventArgs $args)
+    {
+        $repoClass = $args->getRequest()->get('_repositoryClass');
+        $entityId = $args->getRequest()->get('id');
+
+        /** @var Connection $connection */
+        $connection = $args->getSubject()->get('dbal_connection');
+
+        if (strtolower($args->getRequest()->getActionName()) !== 'deletevalues') {
+            return;
+        }
+
+        if ($repoClass !== 'tax') {
+            return;
+        }
+
+        $snippet = $args->getSubject()->get('snippets')->getNamespace('backend/swag_tax/main');
+
+        $productUsed = $connection->fetchColumn('SELECT 1 FROM s_articles WHERE taxID = ?', [$entityId]);
+        $dispatchUsed = $connection->fetchColumn('SELECT 1 FROM s_premium_dispatch WHERE tax_calculation = ?', [$entityId]);
+        $voucherUsed = $connection->fetchColumn('SELECT 1 FROM s_emarketing_vouchers WHERE taxconfig = ?', [$entityId]);
+
+        if ($productUsed !== false) {
+            throw new \RuntimeException($snippet->get('still_used/product'));
```

Comment:
Wouldn't it be better to collect all messages first?

Reply:
I didn't mean it that way. More like: Check for all errors first, and then throw an exception